### DISCUSSION TITLE
Why is Get- Date formatting not unified? Such a mess

### DISCUSSION THREAD

**Discussion by JGFMK** [PowerShell/PowerShell#18662] [Enhancements]
No milliseconds for the UFormat
No AM/PM for Format

https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/get-date?source=recommendations&view=powershell-7.3

You want one that does it all - especially when dealing with MS.

**Comment by lpatalas** [comment]
`-Format` uses [.NET format strings](https://learn.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings) so it should contain everything you need. You can get AM/PM by using `tt`, e.g. `Get-Date -Format tt`.

`-UFormat` on the other hand is missing `%N` specifier for nanoseconds that could be then used to get milliseconds. I guess it's an omission.
